repose-aggregator/src/docs/asciidoc/recipes/docker.adoc

```diff
@@ -1,130 +1,147 @@
+The latest version of the Dockerfiles used to build the *Repose Docker* images published on Docker Hub can be found at:
 
-----
-docker ps
-----
+- https://raw.githubusercontent.com/rackerlabs/repose/master/repose-aggregator/artifacts/docker/src/docker/ubuntu/Dockerfile[Debian/Ubuntu]
+- https://raw.githubusercontent.com/rackerlabs/repose/master/repose-aggregator/artifacts/docker/src/docker/centos/Dockerfile[RedHat/Centos]
 
-=== Inspecting
-Once running, you may wish to inspect the internal state of the container.
-Well, the default command for the official Repose Docker images will run Repose in the foreground, and as a result, simply attaching to the container with a command like the following will only allow interaction with the Repose process:
+To build a custom *Repose Docker* image, follow these steps:
 
+. Acquire a *Repose* Dockerfile.
+.. Clone/Fork the *Repose* project.
+.. cURL/Wget one of the files linked above.
+. *Optionally* Modify the Dockerfile.
+. Run the following command from the root of the *Repose* project to build an image based on the Dockerfile for Ubuntu that will provide *Repose* v8.7.0.2:
++
 ----
-docker attach container_name_or_id
+docker build                                             \ <1>
+   --build-arg REPOSE_VERSION=8.7.0.2                    \ <2>
+   --tag local/repose:v8.7.0.2-ubuntu                    \ <3>
+   ./repose-aggregator/artifacts/docker/src/docker/ubuntu/ <4>
 ----
+Let's break that command down and take a closer look at what it is doing:
++
+<1> This tells Docker to build an image.
+<2> This option is used by the Dockerfile to know what version of *Repose* to install.
+    See our http://www.openrepose.org/versions/latest/release-notes.html[Release Notes] page for a list of available versions.
+<3> This option defines the tag to apply to the image.
+    It is in `repository:tag` format.
+<4> The last parameter is the path to the *Repose* Dockerfile.
+    The Dockerfile itself must exist as a child of the provided directory.
++
+[NOTE]
+====
+This directory will also be treated as the Docker build context.
+====
+. If the Dockerfile was successfully built, a Docker image should be available within Docker.
+Run the following command to show all available images:
++
+----
+docker images
+----
+
+== Building a Repose Docker Image With the build file
+The *Repose* build system expects your Docker Hub credentials to be available at build time.
+They can be passed in as build properties or the easier way is to add them to your `~/.gradle/gradle.properties` file:
 
-Instead, you can use a command like the following to run a new bash session inside of your running container:
+- `dockerhub.username=<USERNAME>`
+- `dockerhub.password=<PASSWORD>`
+
+Then try to build a quick *Repose Docker* image from the root of the *Repose* project.
 
 ----
-docker exec -it container_name_or_id /bin/bash
+./gradlew \
+    :repose-aggregator:artifacts:docker:buildUbuntuImageLocal \
+    -Prepose-version=8.7.0.2
```

Comment:
If we did keep a README around for the Docker project, I might put this in there. Will a Repose user ever need to build a Docker image using the Repose build system? This seems like developer information.

Reply:
I thought it was a logical next step to anyone that was building their own images to see how we build it. That way they can incorporate it into their own build system. Basically it was a way to say, "hey you shouldn't be building it by hand, so here's a better way."

Reply:
Only if their build system is Gradle.

Honestly, I also wouldn't call this a better way. It's an automated way that fits in the build system of Repose, but it:
1. Relies on a Gradle plugin that has a history of giving us trouble.
1. Relies on a Gradle plugin that is pinned to a certain version of the Docker server API, and thus can get out-of-date. This is particularly annoying if you are trying to build an older Repose `Dockerfile` which puts you on a branch relying on an older version of the Gradle plugin.
1. Requires another tool (i.e., Gradle).
1. Reduces flexibility (e.g., you cannot tag an image when building it -- that's another step).

So yes, we benefit from using it because it is easy to incorporate in our release process, but if you aren't releasing Repose, I'm not sure why you would want to use it.

Reply:
I was referring to the building by hand and here is an automated way. The better being automated, not necessarily exactly our way.

Reply:
But if we do move this out to a README, then I'm good with that too.

Reply:
I see. Yeah, I think this totally fits in a README since it's our system, I just can't come up with a reason why someone working with Repose would need to know how to do this. To be fair, the Docker build process is pretty automatic, you're just replacing a single Docker command with a single Gradle command. One, I think, is just clearer (Docker). Also, as far as I know, we haven't publicly documented other Gradle tasks.

Reply:
This was moved into the new developer docs.